iceoryx_posh/test/moduletests/test_roudi_service_registry.cpp

```diff
@@ -29,7 +29,96 @@ namespace
 {
 using namespace ::testing;
 using namespace iox::roudi;
+using iox::capro::ServiceDescription;
+using SearchResult_t = iox::roudi::ServiceRegistry::ServiceDescriptionVector_t;
 
+struct PublisherTest
+{
+    auto add(const ServiceDescription& sd)
+    {
+        return registry.addPublisher(sd);
+    }
+
+    auto otherAdd(const ServiceDescription& sd)
```

Comment:
What about `complementaryAdd()`?

Reply:
Would that gain much? It is not more expressive in my book. It is also an internal artifact to avoid code duplication, no public API.